packages/okta-adapter/src/filters/app_user_schema_deployment.ts

```diff
@@ -0,0 +1,201 @@
+  const { fieldsToOmit } = configUtils.getTypeTransformationConfig(
+    createdAppUserSchemaInstance.elemID.typeName,
+    apiDefinitions.types,
+    apiDefinitions.typeDefaults,
+  )
+  createdAppUserSchemaInstance.value = _.omit(autoCreatedappUserSchema, [
+    ...(fieldsToOmit ?? []).map(field => field.fieldName),
+    'name',
+  ])
+  return createdAppUserSchemaInstance
+}
+
+const makeModificationFromAddition = async (
+  change: AdditionChange<InstanceElement>,
+  client: OktaClient,
+  apiDefinitions: OktaSwaggerApiConfig,
+): Promise<ModificationChange<InstanceElement>> => {
+  const appUserSchemaInstance = getChangeData(change)
+  const parentApplicationId = getParents(appUserSchemaInstance)[0]?.id
+  if (parentApplicationId === undefined) {
+    log.error(`Error while trying to get parent id for user schema ${appUserSchemaInstance.elemID.getFullName()}`)
+    throw new Error(
+      `Could not find parent application id for user schema ${appUserSchemaInstance.elemID.name} from type ${appUserSchemaInstance.elemID.typeName}`,
+    )
+  }
+
+  const autoCreatedAppUserSchema = await getAutoCreatedAppUserSchema(parentApplicationId, client)
+
+  // Assign the id created by the service to the app user schema
+  appUserSchemaInstance.value.id = autoCreatedAppUserSchema.id
+
+  const autoCreatedAppUserSchemaInstance = getAppUserSchemaInstance(
+    autoCreatedAppUserSchema,
+    appUserSchemaInstance,
+    apiDefinitions,
+  )
+  return { action: 'modify', data: { before: autoCreatedAppUserSchemaInstance, after: appUserSchemaInstance.clone() } }
+}
+
+const deployModificationChange = async (
+  change: ModificationChange<InstanceElement>,
+  client: OktaClient,
+  apiDefinitions: OktaSwaggerApiConfig,
+): Promise<void> => {
+  const { before } = change.data
+  const after = change.data.after.clone()
+  if (before.value[DEFINITIONS]?.[BASE] !== undefined) {
+    _.set(after.value, [DEFINITIONS, BASE], before.value[DEFINITIONS][BASE])
+  } else {
+    _.unset(after.value, [DEFINITIONS, BASE])
+  }
+  await defaultDeployWithStatus(toChange({ before, after }), client, apiDefinitions)
+}
+
+const deployAdditionChange = async (
+  change: AdditionChange<InstanceElement>,
+  client: OktaClient,
+  apiDefinitions: OktaSwaggerApiConfig,
+): Promise<void> => {
+  const modifiedChange = await makeModificationFromAddition(change, client, apiDefinitions)
+  return deployModificationChange(modifiedChange, client, apiDefinitions)
+}
+
+const deployRemovalChange = async (change: Change, client: OktaClient): Promise<void> => {
+  const appUserSchemaInstance = getChangeData(change)
+  const parentApplicationId = getParents(appUserSchemaInstance)[0]?.id
+  if (!_.isString(parentApplicationId) || !(await verifyApplicationIsDeleted(parentApplicationId, client))) {
+    throw new Error('Expected the parent Application to be deleted')
+  }
+}
+
+const deployChange = async (
+  change: Change<InstanceElement>,
+  client: OktaClient,
```

Comment:
```suggestion
client: clientUtils.HTTPWriteClientInterface & clientUtils.HTTPReadClientInterface,
```

Also in all other functions in this file.